### PR TITLE
A couple of fixes for issues reported by Coverity

### PR DESCRIPTION
**Pull request by asimrp**
One in pg_regress and another one in pxf_fdw.

In the pg_regress fix, I couldn't find usage of `--prehook` in code, @pivotal-ning-yu where is it used?

### CODE REVIEW

**Review by frankgh — APPROVED** [review]
The PXF_FDW code looks good

File: src/test/regress/pg_regress.c [hunk 1]
```diff
@@ -2924,19 +2924,20 @@ regression_main(int argc, char *argv[], init_function ifunc, test_function tfunc
 									  strlen(".sql") +
 									  1 /* '\0' */);
 		sprintf(fullname, "%s/sql/hooks/%s.sql", inputdir, prehook);
```
Comment (hlinnaka) [inline src/test/regress/pg_regress.c:2926]: You could use psprintf() here, would be simpler.

File: src/test/regress/pg_regress.c [hunk 2]
```diff
@@ -2924,19 +2924,20 @@ regression_main(int argc, char *argv[], init_function ifunc, test_function tfunc
...
-			if (!file_exists(prehook))
+			if (!file_exists(fullname))
 			{
 				fprintf(stderr, _("%s: could not open file \"%s\" for reading: %s\n"),
-						progname, prehook, strerror(errno));
+						progname, fullname, strerror(errno));
 				exit(2);
 			}
 		}
+		free(prehook);
```
Comment (hlinnaka) [inline src/test/regress/pg_regress.c:2939]: We don't generally bother to free one-off allocations like this in pg_regress and other client tools. The memory is freed when the process exits anyway. In upstream we've just dismissed coverity warnings about memory leaks in such cases.

**Review by hlinnaka — APPROVED** [review]

File: src/test/regress/pg_regress.c (hunk 1 above)
Comment (asimrp) [reply src/test/regress/pg_regress.c:2926]: Indeed, changed as suggested.

File: src/test/regress/pg_regress.c (hunk 2 above)
Comment (asimrp) [reply src/test/regress/pg_regress.c:2939]: Removed the call to free() to keep it consistent with upstream.

**Review by asimrp — COMMENTED** [review]
Pushed to master 325a13980ec..905814cb6f7, thanks for the reviews.

**Pull request closed without merging**